lms/djangoapps/certificates/models.py

```diff
@@ -1063,7 +1063,7 @@ def __unicode__(self):
 
     class Meta(object):
         get_latest_by = 'created'
-        unique_together = (('organization_id', 'course_key', 'mode'),)
+        unique_together = (('organization_id', 'course_key', 'mode', 'language'),)
```

Comment:
('edx', 'this_is_a_course_id', 'audit') would be considered unique before the migration.
('edx', 'this_is_a_course_id', 'audit') would not be considered unique after the migration.
This is okay going forward.

Let's say we need to roll back and these entries have been added.
('edx', 'this_is_a_course_id', 'audit', 'es') 
('edx', 'this_is_a_course_id', 'audit', 'en')
These two are now duplicates and not unique so I'm not sure if it can be rolled back.

Reply:
@BenjiLee I agree rolling back would be a problem after new templates with different languages. I'm not sure of how to mitigate this.

Reply:
@edx/learner-heart know of a solution? I believe altering this unique_together is unavoidable if we want to enable spanish templates.

Reply:
I would search for any templates that have matching `('organization_id', 'course_key', 'mode')`  and delete the spanish one (aka the new one).  Or something along those lines.  If we were really paranoid we could log the data in the deleted rows so that we have a record of what was in there.

Reply:
Anywhere specific I should log the data on deleted rows to?

Reply:
What would happen if we ran the reverse migration without removing any data? Would the old restraint prevent us from doing so and cause the reversal to just fail? Maybe that's better in an opensource environment than removing data as it would force someone to look at and manually fix the data before this migration could be run.

Reply:
^Yes I believe the migration would fail if we didnt remove any data, I pushed up a version that has a reversal (which would delete a language-specific templates if there are multiple templates for that course), I haven't tested it a ton yet but give it a look

Reply:
I don't know that we should delete data on reversal. I think it might be better to let it fail in that case and force the admin to decide what to do with the duplicate data, especially since this is being consumed in an open source environment and it may be unexpected for the reversal of this migration to remove data from the database. @bderusha, @BenjiLee, thoughts?

Reply:
I stand by my last comment.  The migration above all else.  The migration cannot be allowed to fail during a rollback.  If we are rolling back the migration it is because we need the database back in its original state.

Losing data that could not have existed before is part of that price.  I think we should log any deleted data so that the manual recover later is easier, but we should not fail the rollback to save the data.